project/src/main/kotlin/org/ionproject/core/accessControl/PDP.kt

```diff
@@ -17,8 +21,64 @@ private val logger = LoggerFactory.getLogger(LoggerInterceptor::class.java)
 @Component
 class PDP {
     private val pap: AuthRepoImpl = AuthRepoImpl(TransactionManagerImpl(DataSourceHolder))
+    private val tokenGenerator = TokenGenerator()
 
-    fun evaluateRequest(tokenHash: String, requestDescriptor: Request): Boolean {
+    /**
+     * Evaluates if the access token received as a query parameter is valid
+     */
+    fun evaluateJwtToken(jwt: String, method: String, requestUrl: String) {
+        val parts = jwt.split(".")
+
+        //Check method - this type of access doesn't allow UNSAFE METHODS (POST, PUT...)
+        if(method != "GET" && method != "HEAD") {
+            logger.info("An access_token query parameter authentication method failed: INVALID METHOD \"$method\" ")
+            throw BadRequestException("This type of access doesn't allow unsafe methods (PUT, POST...).")
+        }
```

Comment:
Http Method check could be before split. Just a small optimization.

Reply:
Agreed